src/stm32_generic/quadrature_encoder.cpp

```diff
@@ -74,19 +74,21 @@ void setup_enable_register(int channel, timer_reg_t* p_reg)
 
 quadrature_encoder::quadrature_encoder(hal::unsafe,
                                        encoder_channels channels,
-                                       void* p_reg)
+                                       void* p_reg,
+                                       float p_pulses_per_rotation)
 {
-  initialize(hal::unsafe{}, channels, p_reg);
+  initialize(hal::unsafe{}, channels, p_reg, p_pulses_per_rotation);
 }
 quadrature_encoder::quadrature_encoder(hal::unsafe)
 {
 }
 void quadrature_encoder::initialize(unsafe,
                                     encoder_channels channels,
-                                    void* p_reg)
+                                    void* p_reg,
+                                    float p_pulses_per_rotation)
```

Comment:
In general, the number will usually be a whole number and not a fraction. Lets accept this value as a u32.